web/src/main/java/com/bazaarvoice/emodb/web/resources/blob/BlobStoreResource1.java

```diff
@@ -109,13 +108,12 @@ public BlobStoreResource1(BlobStore blobStore, DataCenters dataCenters, Set<Stri
     public Iterator<Table> listTables(@QueryParam("from") final String fromKeyExclusive,
                                       @QueryParam("limit") @DefaultValue("10") LongParam limit,
                                       final @Authenticated Subject subject) {
-        final Iterator<Table> tables = _blobStore.listTables(Strings.emptyToNull(fromKeyExclusive), limit.get());
-        final UnmodifiableIterator<Table> permittedTables = Iterators.filter(tables, new Predicate<Table>() {
-            @Override public boolean apply(final Table input) {
-                return subject.hasPermission(Permissions.readBlobTable(new NamedResource(input.getName())));
-            }
-        });
-        return streamingIterator(permittedTables);
+        return streamingIterator(
+            StreamSupport.stream(((Iterable<Table>) () -> _blobStore.listTables(Strings.emptyToNull(fromKeyExclusive), Long.MAX_VALUE)).spliterator(), false)
```

Comment:
Not asking that you change it, but we've been more commonly using `Spliterators.spliteratorUnknownSize()` to go from iterators to streams.  For example:

```
StreamSupport.stream(Spliterators.spliteratorUnknownSize(_blobStore.listTables(Strings.emptyToNull(fromKeyExclusive), Long.MAX_VALUE), 0), false)
```

It's no shorter, a valid argument can be made for either one being more readable, and works exactly the same, so take it or leave it, just making you aware of the alternative if you didn't already know about it.